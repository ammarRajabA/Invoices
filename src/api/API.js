import axios from 'axios';
import config from '../config'

export const API=axios.create({
	baseURL: config.backend,
	headers: {
		'Content-Type': 'application/json'
	}
});
