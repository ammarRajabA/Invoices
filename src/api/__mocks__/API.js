export const API={
	get: jest.fn(() => Promise.resolve({ data: "mocked" })),
	put: jest.fn(() => Promise.resolve({ data: "mocked" })),
	delete: jest.fn(() => Promise.resolve({ data: "mocked" })),
	post: jest.fn(() => Promise.resolve({ data: "mocked" }))
}
