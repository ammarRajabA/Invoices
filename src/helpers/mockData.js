import faker from 'faker';

function randomInvoice() {
  const senderName = `${faker.name.firstName()} ${faker.name.lastName()}`;
  const recipientName = `${faker.name.firstName()} ${faker.name.lastName()}`;
  return {
    "id": faker.random.uuid(),
    "dueDate": faker.date.future(),
    "createdDate": faker.date.past(),
    "sender": {
      "name": senderName,
      "email": `${senderName.split(" ").map(n => n.toLowerCase()).join(".")}@sender.com`,
      "address1": `${faker.company.companyName()} ${faker.company.companySuffix()}`,
      "address2": `${faker.address.streetName()}, ${faker.random.number({min: 1, max: 100})}`,
      "address3": `${faker.address.zipCode()} ${faker.address.city()}`,
      "vatId": faker.helpers.replaceSymbolWithNumber("DE#########")
    },
    "recipient": {
      "name": recipientName,
      "email": `${recipientName.split(" ").map(n => n.toLowerCase()).join(".")}@recipient.com`,
      "address1": `${faker.company.companyName()} ${faker.company.companySuffix()}`,
      "address2": `${faker.address.streetName()}, ${faker.random.number({min: 1, max: 100})}`,
      "address3": `${faker.address.zipCode()} ${faker.address.city()}`,
      "vatId": faker.helpers.replaceSymbolWithNumber("DE#########")
    },
    "items": Array.apply(null, new Array(faker.random.number({min: 1, max: 20}))).map(() => (
      {
        "description": faker.commerce.productName(),
        "qty": faker.random.number({min: 1, max: 10}),
        "unitPriceNet": parseFloat(faker.finance.amount(0.01,100,2)),
        "taxRate": 0.19
      }
    ))
  }
}

const NUM_INVOICES = 10;
const INVOICES = Array.apply(null, new Array(NUM_INVOICES)).map(randomInvoice);

export default {
	invoices:INVOICES,
	meta:{
		total:100,
		totalPages:10,
		pageNumber:1,
		pageSize:10
	},
  INIT_STATE:{
    invoices:[],
    selectedInvoice:null,
    loading:false,
    invoiceSelected:null,
    meta:{
      total:0,
      totalPages:0,
      pageNumber:0,
      pageSize:0
    },
    newInvoice:{
      "id": -1,
        "dueDate": new Date(),
        "createdDate": new Date(),
        "sender": {
            "name": '',
            "email": '',
            "address1": '',
            "address2": '',
            "address3": '',
            "vatId": ''
        },
        "recipient": {
            "name": '',
            "email": '',
            "address1": '',
            "address2": '',
            "address3": '',
            "vatId": ''
        },
        "items": [{description:'',qty:0,unitPriceNet:0,taxRate:0}]
    }
  }
}