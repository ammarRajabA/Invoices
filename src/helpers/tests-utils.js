import React from 'react'
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

export const makeMountRender = (Component, defaultProps = {}) => {
	return (customProps={})=>{
		const props={...defaultProps,...customProps};
		return mount(<Component {...props}/>)
	}
}

export const reduxify = (Component, props = {}, store= {}) => {
	return function reduxWrap() {
		return (
			<Provider store={store}>
				<Component {...props} />
			</Provider>
		)
	}
}

export const snapshotify = reactWrapper => {
	return reactWrapper.debug();
};