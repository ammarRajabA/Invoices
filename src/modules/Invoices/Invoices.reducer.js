const INIT_STATE={
	invoices:[],
	selectedInvoice:null,
	loading:false,
	invoiceSelected:null,
	meta:{
		total:0,
		totalPages:0,
		pageNumber:0,
		pageSize:0
	},
	newInvoice:{
		"id": -1,
    	"dueDate": new Date(),
    	"createdDate": new Date(),
    	"sender": {
      		"name": '',
      		"email": '',
      		"address1": '',
      		"address2": '',
      		"address3": '',
      		"vatId": ''
    	},
    	"recipient": {
      		"name": '',
      		"email": '',
      		"address1": '',
      		"address2": '',
      		"address3": '',
      		"vatId": ''
    	},
    	"items": [{description:'',qty:0,unitPriceNet:0,taxRate:0}]
	}
}

export default (state=INIT_STATE,action)=>{
	switch(action.type){
		case 'STARTED_FETCH_INVOICES':
			return {...state,loading:true};
		case 'STARTED_SAVING_INVOICES':
			return {...state,loading:true};
		case 'FINISHED_FETCH_INVOICES':
			return {...state,invoices:action.payload.invoices,meta:action.payload.meta,loading:false};
		case 'FAILED_FETCH_INVOICES':
			return {...state}
		case 'SELECT_INVOICE':
			return {...state,invoiceSelected:action.payload}
		case 'UPDATE_INVOICE':
			var updatedInvoice={...state.invoiceSelected};
			updatedInvoice[action.payload.key]=action.payload.value
			return {...state,invoiceSelected:updatedInvoice}
		case 'NEW_INVOICE':
			var newInvoice={...state.newInvoice}
			return {...state,invoiceSelected:newInvoice}
		case 'SORT_INVOICES':
			return {...state,invoices:action.payload,selectedInvoice:null}
		default:
			return state
	}
}