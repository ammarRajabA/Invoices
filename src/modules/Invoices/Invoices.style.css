.container{
	width: 100%;
	height: 100%;
	background-color: #FFFFFF;
}
.header{
	width: 90%;
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: space-between;
	border-bottom: 1px solid #EFEFEF;
	padding-bottom: 5px;
	margin-bottom: 20px;
}
.header .title{
	margin-left: 20px;
	font-family: calibri;
	cursor: default;
}
.labels {
	width: 90%;
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: flex-start;
	margin-bottom: 15px
}
.card{
	margin-right: auto;
	margin-left: auto;
	width: 90%;
	padding: 20px;
	border-radius: 5px;
	background-color: #FFFFFF;
	box-shadow: 0 1px 1px 0 rgba(0,0,0,0.2), 0 3px 10px 0 rgba(0,0,0,0.19);
	overflow-y: scroll;
	overflow-x: scroll;
	font-family: calibri;
}
.pagination{
	position: relative;
	margin-top: 20px;
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: center;
	font-size: 16px;
}
.pagination .page-size{
	position: absolute;
	display:flex;
	flex-direction:row;
	align-items:center;
	right: 10px;
}
.name{
	color:#2643F7;
	font-weight:bold
}
.tooltip{
	position: absolute;
	opacity: 0;
	background-color: #313131;
	transition: 0.3s;
	color: white;
	padding: 10px;
	border-radius: 10px;
	font-family: calibri;
	font-size: 14px;
	right: 0px;
	bottom: -15px;
}
.tooltip:hover{
	opacity: 0.9;
}