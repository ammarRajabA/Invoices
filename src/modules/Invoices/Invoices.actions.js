import {API} from '../../api/API';

export const getInvoices=(page_number=1,page_size=10)=>async dispatch=>{
	try{
		dispatch({type:"STARTED_FETCH_INVOICES"})
		var response=await API.get(`/invoices?page_size=${page_size}&page_number=${page_number}`);
		if (response.status===200){
			dispatch({type:"FINISHED_FETCH_INVOICES",payload:response.data})
		}else{
			dispatch({type:"FAILED_FETCH_INVOICES"})
		}
	}catch(err){
		dispatch({type:"FAILED_FETCH_INVOICES"})
	}
}

export const sortInvoices=(invoices)=>{
	return ({type:'SORT_INVOICES',payload:invoices})
}

export const saveInvoice=(invoice)=>async dispatch=>{
	try{
		dispatch({type:"STARTED_SAVING_INVOICES"})
		if (invoice.id<0)
			var response=await API.post(`/invoices`,{invoice});
		else
			var response=await API.put(`/invoices/${invoice.id}`,{invoice});
		getInvoices()(dispatch);
	}catch(err){
		
	}
}

export const deleteInvoice=(invoice)=>async dispatch=>{
	try{
		dispatch({type:"STARTED_FETCH_INVOICES"})
		var response=await API.delete(`/invoices/${invoice.id}`);
		getInvoices()(dispatch)
	}catch(err){
		
	}
}

export const selectInvoice =(invoice)=>{
	return {type:'SELECT_INVOICE',payload:invoice}
}
export const newInvoice =()=>{
	return {type:'NEW_INVOICE'}
}

export const updateInvoice=(key,value)=>{
	return {type:'UPDATE_INVOICE',payload:{key,value}}
}