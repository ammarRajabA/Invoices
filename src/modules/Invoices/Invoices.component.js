import React, { Component } from 'react';
import { connect } from 'react-redux';
import moment from 'moment'

import {Button,Dropdown,Input,Table,Spinner,Search,PopupMenu,Confirm,Label} from "../../components";
import Invoice from '../Invoice/Invoice.component'

import {calculateTotalPriceWithTax} from '../../helpers/finance'

import {getInvoices,selectInvoice,newInvoice,deleteInvoice,sortInvoices} from './Invoices.actions'

import './Invoices.style.css'

class Invoices extends Component{
	constructor(props){
		super(props);
		this.state={page_size:10,invoiceHover:null,invoiceOptions:null,showMenu:false,showInvoice:false,editMode:false,showConfirm:false,sort:{column:null,asc:true},searchText:''}
	}
	componentDidMount(){
		this.props.getInvoices()
	}
	componentDidUpdate(prevProps){
		if (prevProps.invoicesReducer.loading!==this.props.invoicesReducer.loading && !this.props.invoicesReducer.loading){
			if (this.state.sort.column){
				var invoices=[...this.props.invoicesReducer.invoices];
				invoices.sort((a,b)=>a[this.state.sort.column].localeCompare(b[this.state.sort.column]))
				if (!this.state.sort.asc) invoices.reverse();
				this.props.sortInvoices(invoices);
			}
			this.setState({page_size:this.props.invoicesReducer.meta.pageSize})
		}
	}
	onMouseOverInvoice=(e,index)=>{
		this.setState({invoiceHover:index})
	}
	refreshPage=()=>{
		this.props.getInvoices(this.props.invoicesReducer.meta.pageNumber,this.state.page_size)
	}
	nextPage=()=>{
		this.props.getInvoices(this.props.invoicesReducer.meta.pageNumber+1,this.state.page_size)
	}
	prevPage=()=>{
		this.props.getInvoices(this.props.invoicesReducer.meta.pageNumber-1,this.state.page_size)
	}
	firstPage=()=>{
		this.props.getInvoices(1,this.state.page_size)
	}
	lastPage=()=>{
		this.props.getInvoices(this.props.invoicesReducer.meta.totalPages,this.state.page_size)
	}
	handleOption=(option,index)=>{
		switch(option){
			case 'Edit':
				this.setState({editMode:true,showInvoice:true},()=>{
					this.props.selectInvoice(this.props.invoicesReducer.invoices[this.state.invoiceOptions])
				})
				break;
			case 'Details':
				this.setState({editMode:false,showInvoice:true},()=>{
					this.props.selectInvoice(this.props.invoicesReducer.invoices[this.state.invoiceOptions])
				})
				break;
			case 'Delete':
				this.setState({showConfirm:true},()=>{
					this.props.selectInvoice(this.props.invoicesReducer.invoices[this.state.invoiceOptions])
				})
				break;
			default:
				return;
		}
	}
	onSort=(event,key)=>{
		if (!key) return
		var asc=this.state.sort.column?(this.state.sort.asc?false:true):true
		var invoices=[...this.props.invoicesReducer.invoices];
		invoices.sort((a,b)=>a[key].localeCompare(b[key]))
		if (!asc) invoices.reverse();
		this.props.sortInvoices(invoices)
		this.setState({sort:{column:key,asc:asc}})
	}
	search=(invoice)=>{
		return invoice.id.toString().indexOf(this.state.searchText)>=0
	}
	
	renderHeader=()=>{
		return (
			<div className="header">
				<h1 className="title">Invoices</h1>
				<Search placeholder="Invoice #" value={this.state.searchText} onChange={(e)=>this.setState({searchText:e.target.value})}/>
				<Button icon="fas fa-plus" label="Add Invoice" onClick={
					()=>this.setState({editMode:true,showInvoice:true},()=>{
						this.props.newInvoice()
					})
				}/>
			</div>
			)
	}
	renderLabels=()=>{
		if (this.props.invoicesReducer.loading) return <span></span>
		return(
			<div className="labels">
				<Label title={calculateTotalPriceWithTax(this.props.invoicesReducer.invoices).toFixed(2)+" $"} subtitle="Below invoices total" minilabels={[{title:this.props.invoicesReducer.meta.total,subtitle:"All invoices"},{title:this.props.invoicesReducer.meta.pageSize,subtitle:"Below invoices"}]}/>
			</div>
			)
	}
	renderPagination=()=>{
		return (
				<div className="pagination">
					<Button type={(this.props.invoicesReducer.meta.pageNumber===1)?'disabled':'primary'} icon="fas fa-angle-double-left" onClick={this.firstPage} style={{width:40,height:40,borderRadius:20,fontSize:12}}/>
					<Button type={(this.props.invoicesReducer.meta.pageNumber===1)?'disabled':'primary'} icon="fas fa-chevron-left" onClick={this.prevPage} style={{width:50,height:50,borderRadius:25}}/>
					<span>{this.props.invoicesReducer.meta.pageNumber}</span>
					<Button type={(this.props.invoicesReducer.meta.pageNumber===this.props.invoicesReducer.meta.totalPages)?'disabled':'primary'} icon="fas fa-chevron-right" onClick={this.nextPage} style={{width:50,height:50,borderRadius:25}}/>
					<Button type={(this.props.invoicesReducer.meta.pageNumber===this.props.invoicesReducer.meta.totalPages)?'disabled':'primary'} icon="fas fa-angle-double-right" onClick={this.lastPage} style={{width:40,height:40,borderRadius:20,fontSize:12}}/>
					<span className="page-size">
						<Input placeholder="Page Size" type="number" onChange={(e)=>this.setState({page_size:e.target.value})} value={this.state.page_size} style={{width:50,textAlign:'center'}}/>
						<Button icon="fas fa-sync" onClick={this.refreshPage} style={{width:40,height:40,borderRadius:20,fontSize:12}}/>
					</span>
				</div>
			)
	}
	renderInvoices=()=>{
		if (this.props.invoicesReducer.loading) return <Spinner message="Loading..."/>
		return (
			<div className="card">
				<Table onMouseLeave={()=>this.setState({invoiceHover:null})}>
					<tbody>
						<tr>
							<th>Invoice #</th>
						    <th onClick={e=>this.onSort(e,'dueDate')}>
						    	<span>Due Date</span>
						    	{
						    		(this.state.sort.column==='dueDate')?
						    			((this.state.sort.asc)?
						    			<i style={{marginLeft:20}} className="fas fa-sort-amount-up"></i>:
						    			<i style={{marginLeft:20}} className="fas fa-sort-amount-down"></i>):
						    			<i style={{marginLeft:20,color:'#B1B1B1'}} className="fas fa-sort-amount-down"></i>
						    	}
						    </th>
						    <th onClick={e=>this.onSort(e,'createdDate')}>
						    	<span>Created Date</span>
						    	{
						    		(this.state.sort.column==='createdDate')?
						    			((this.state.sort.asc)?
						    			<i style={{marginLeft:20}} className="fas fa-sort-amount-up"></i>:
						    			<i style={{marginLeft:20}} className="fas fa-sort-amount-down"></i>):
						    			<i style={{marginLeft:20,color:'#B1B1B1'}} className="fas fa-sort-amount-down"></i>
						    	}
						    </th>
						    <th>Sender Name</th>
						    <th>Recipient Name</th> 
						    <th>Total Price (with tax)</th>
						    <th style={{width:80}}></th>
					  	</tr>
						{
							this.props.invoicesReducer.invoices.filter(this.search).map((invoice,index)=>
								<tr onMouseOver={(e)=>this.onMouseOverInvoice(e,index)} key={invoice.id.toString()} className="invoice-row">
								    <td>{invoice.id}</td>
								    <td>{moment(invoice.dueDate).format('lll')}</td>
								    <td>{moment(invoice.createdDate).format('lll')}</td> 
								    <td className="name">{invoice.sender.name}<span className="tooltip">{invoice.sender.email}</span></td>
								    <td>{invoice.recipient.name}<span className="tooltip">{invoice.recipient.email}</span></td>
								    <td>{calculateTotalPriceWithTax(invoice).toFixed(2)} $</td>
								    <td>
								    	<Button onClick={()=>this.setState({showMenu:true,invoiceOptions:index})} icon="fas fa-ellipsis-v" type={(this.state.invoiceHover===index)?"primary":"solid"} style={{width:40,height:40,borderRadius:20}}/>
								    	<PopupMenu
								    		options={['Edit','Details','Delete']}
								    		handleOption={this.handleOption}
								    		show={this.state.showMenu && this.state.invoiceOptions===index}
								    		onDismiss={()=>this.setState({showMenu:false,invoiceOptions:null})}
								    	/>
								    </td>
								</tr>
							)
						}
					</tbody>
				</Table>
				{this.renderPagination()}
			</div>
			)
	}
	renderInvoice=()=>{
		return <Invoice editMode={this.state.editMode} show={this.state.showInvoice} onDismiss={()=>this.setState({showInvoice:false})}/>
	}
	renderConfirm=()=>{
		return <Confirm title="Delete Invoice" message="You are about to delete invoice, continue ?"
						onConfirm={()=>this.setState({showConfirm:false},()=>this.props.deleteInvoice(this.props.invoicesReducer.invoiceSelected))}
						onCancel={()=>this.setState({showConfirm:false})}
						onDismiss={()=>this.setState({showConfirm:false})}
						show={this.state.showConfirm} type="danger"
				/>
	}
	render(){
		return(
				<div className="container">
					{this.renderHeader()}
					{this.renderLabels()}
					{this.renderInvoices()}
					{this.renderInvoice()}
					{this.renderConfirm()}
				</div>
			)
	}
}

const mapStateToProps=(state)=>{
	return ({
		invoicesReducer:state.invoicesReducer
	})
}

export default connect(mapStateToProps,{getInvoices,selectInvoice,newInvoice,deleteInvoice,sortInvoices})(Invoices)