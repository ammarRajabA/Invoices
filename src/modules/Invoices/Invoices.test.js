import {API} from '../../api/API'
import configureMockStore from "redux-mock-store";
import ReduxThunk from "redux-thunk";
import { makeMountRender, snapshotify, reduxify } from '../../helpers/tests-utils';

import Invoices from './Invoices.component';
import {getInvoices} from './Invoices.actions';
import mockData from '../../helpers/mockData'

jest.mock('../../api/API')

const mockStore = configureMockStore([ReduxThunk]);

describe('Testing getInvoices action creator',function(){
	it ("dispatch FAILED_FETCH_INVOICES action on error", async ()=>{
		let store= mockStore({invoicesReducer:mockData.INIT_STATE});
		API.get.mockImplementationOnce(() =>Promise.resolve({status:500}))
		await store.dispatch(getInvoices());
		const actions = store.getActions();

		expect.assertions(2);
		expect(actions[0].type).toEqual("STARTED_FETCH_INVOICES");
    	expect(actions[1].type).toEqual("FAILED_FETCH_INVOICES");
	})
	it ("dispatch FINISHED_FETCH_INVOICES action and return data on success", async ()=>{
		let store= mockStore({invoicesReducer:mockData.INIT_STATE});
		API.get.mockImplementationOnce(() =>Promise.resolve({status:200,data:{invoices:mockData.invoices,meta:mockData.meta}}))
		await store.dispatch(getInvoices());
		const actions = store.getActions();

		expect.assertions(3);
		expect(actions[0].type).toEqual("STARTED_FETCH_INVOICES");
    	expect(actions[1].type).toEqual("FINISHED_FETCH_INVOICES");
		expect(actions[1].payload.invoices.length).toEqual(10);
	})
})

describe('Invoices component',function () {
	it('match snapshot',()=>{
		let store= mockStore({invoicesReducer:mockData.INIT_STATE});
		var wrapper = makeMountRender(reduxify(Invoices,{},store))();
		wrapper=wrapper.update()
		expect(snapshotify(wrapper)).toMatchSnapshot();
	})
	it('renders list of invoices that matches pageSize',()=>{
		let store= mockStore({invoicesReducer:{...mockData.INIT_STATE,invoices:mockData.invoices,meta:mockData.meta}});
		var wrapper = makeMountRender(reduxify(Invoices,{},store))();
		var rows=wrapper.find('tr');
		var pageSize=wrapper.find('input').last().props().value;
		expect(rows.length).toEqual(pageSize+1)
	})
})