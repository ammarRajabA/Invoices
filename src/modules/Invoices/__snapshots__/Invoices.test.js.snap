// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoices component match snapshot 1`] = `
"<reduxWrap>
  <Provider store={{...}}>
    <Connect(Invoices)>
      <Invoices invoicesReducer={{...}} getInvoices={[Function]} selectInvoice={[Function]} newInvoice={[Function]} deleteInvoice={[Function]} sortInvoices={[Function]}>
        <div className=\\"container\\">
          <div className=\\"header\\">
            <h1 className=\\"title\\">
              Invoices
            </h1>
            <Input placeholder=\\"Invoice #\\" value=\\"\\" onChange={[Function: onChange]}>
              <span className=\\"search\\">
                <input placeholder=\\"Invoice #\\" value=\\"\\" onChange={[Function: onChange]} />
              </span>
            </Input>
            <Button icon=\\"fas fa-plus\\" label=\\"Add Invoice\\" onClick={[Function: onClick]}>
              <button className=\\"button undefined\\" icon=\\"fas fa-plus\\" label=\\"Add Invoice\\" onClick={[Function: onClick]}>
                <i className=\\"fas fa-plus\\" />
                <span>
                  Add Invoice
                </span>
              </button>
            </Button>
          </div>
          <div className=\\"labels\\">
            <Label title=\\"0.00 $\\" subtitle=\\"Below invoices total\\" minilabels={{...}}>
              <span className=\\"label-component undefined\\" title=\\"0.00 $\\" subtitle=\\"Below invoices total\\" minilabels={{...}}>
                <span className=\\"label-subtitle\\">
                  Below invoices total
                </span>
                <span className=\\"label-title\\">
                  0.00 $
                </span>
                <span className=\\"label-footer\\">
                  <span className=\\"mini-label\\">
                    <span className=\\"mini-label-subtitle\\">
                      All invoices
                    </span>
                    <span className=\\"mini-label-title\\">
                      0
                    </span>
                  </span>
                  <span className=\\"mini-label\\">
                    <span className=\\"mini-label-subtitle\\">
                      Below invoices
                    </span>
                    <span className=\\"mini-label-title\\">
                      0
                    </span>
                  </span>
                </span>
              </span>
            </Label>
          </div>
          <div className=\\"card\\">
            <Button onMouseLeave={[Function: onMouseLeave]}>
              <table className=\\"table\\" onMouseLeave={[Function: onMouseLeave]}>
                <tbody>
                  <tr>
                    <th>
                      Invoice #
                    </th>
                    <th onClick={[Function: onClick]}>
                      <span>
                        Due Date
                      </span>
                      <i style={{...}} className=\\"fas fa-sort-amount-down\\" />
                    </th>
                    <th onClick={[Function: onClick]}>
                      <span>
                        Created Date
                      </span>
                      <i style={{...}} className=\\"fas fa-sort-amount-down\\" />
                    </th>
                    <th>
                      Sender Name
                    </th>
                    <th>
                      Recipient Name
                    </th>
                    <th>
                      Total Price (with tax)
                    </th>
                    <th style={{...}} />
                  </tr>
                </tbody>
              </table>
            </Button>
            <div className=\\"pagination\\">
              <Button type=\\"primary\\" icon=\\"fas fa-angle-double-left\\" onClick={[Function]} style={{...}}>
                <button className=\\"button primary\\" type=\\"primary\\" icon=\\"fas fa-angle-double-left\\" onClick={[Function]} style={{...}}>
                  <i className=\\"fas fa-angle-double-left\\" />
                </button>
              </Button>
              <Button type=\\"primary\\" icon=\\"fas fa-chevron-left\\" onClick={[Function]} style={{...}}>
                <button className=\\"button primary\\" type=\\"primary\\" icon=\\"fas fa-chevron-left\\" onClick={[Function]} style={{...}}>
                  <i className=\\"fas fa-chevron-left\\" />
                </button>
              </Button>
              <span>
                0
              </span>
              <Button type=\\"disabled\\" icon=\\"fas fa-chevron-right\\" onClick={[Function]} style={{...}}>
                <button className=\\"button disabled\\" type=\\"disabled\\" icon=\\"fas fa-chevron-right\\" onClick={[Function]} style={{...}}>
                  <i className=\\"fas fa-chevron-right\\" />
                </button>
              </Button>
              <Button type=\\"disabled\\" icon=\\"fas fa-angle-double-right\\" onClick={[Function]} style={{...}}>
                <button className=\\"button disabled\\" type=\\"disabled\\" icon=\\"fas fa-angle-double-right\\" onClick={[Function]} style={{...}}>
                  <i className=\\"fas fa-angle-double-right\\" />
                </button>
              </Button>
              <span className=\\"page-size\\">
                <Input placeholder=\\"Page Size\\" type=\\"number\\" onChange={[Function: onChange]} value={10} style={{...}}>
                  <span className=\\"input-container\\">
                    <span className=\\"label \\" style={{...}}>
                      Page Size
                    </span>
                    <input className=\\"input \\" placeholder=\\"Page Size\\" type=\\"number\\" onChange={[Function: onChange]} value={10} style={{...}} />
                  </span>
                </Input>
                <Button icon=\\"fas fa-sync\\" onClick={[Function]} style={{...}}>
                  <button className=\\"button undefined\\" icon=\\"fas fa-sync\\" onClick={[Function]} style={{...}}>
                    <i className=\\"fas fa-sync\\" />
                  </button>
                </Button>
              </span>
            </div>
          </div>
          <Connect(Invoice) editMode={false} show={false} onDismiss={[Function: onDismiss]}>
            <Invoice editMode={false} show={false} onDismiss={[Function: onDismiss]} invoicesReducer={{...}} updateInvoice={[Function]} saveInvoice={[Function]}>
              <span />
            </Invoice>
          </Connect(Invoice)>
          <Confirm title=\\"Delete Invoice\\" message=\\"You are about to delete invoice, continue ?\\" onConfirm={[Function: onConfirm]} onCancel={[Function: onCancel]} onDismiss={[Function: onDismiss]} show={false} type=\\"danger\\">
            <span />
          </Confirm>
        </div>
      </Invoices>
    </Connect(Invoices)>
  </Provider>
</reduxWrap>"
`;
